tensorboard/webapp/metrics/views/card_renderer/utils_test.ts

```diff
@@ -38,4 +39,134 @@ describe('metrics card_renderer utils test', () => {
       ).toBe('eid/foo/bar');
     });
   });
+
+  function buildPoints(xs: number[]): PartialSeries['points'] {
+    return xs.map((x) => {
+      return {x, y: x * 10, value: x * 10, wallTime: 0, step: x};
+    });
+  }
+
+  describe('#partitionSeries', () => {
+    it('partitions series when non-monotonic points are detected', () => {
+      const actual = partitionSeries([
+        {
+          runId: 'a',
+          points: buildPoints([1, 2, 2, 1, 5, 10]),
+        },
+        {
+          runId: 'b',
+          points: buildPoints([1, 1, 1]),
+        },
+      ]);
+
+      expect(actual).toEqual([
+        {
+          seriesId: '["a",0]',
+          runId: 'a',
+          points: buildPoints([1, 2, 2]),
```

Comment:
The more I think about it, the more I wonder whether this should actually be split into [1, 2] and [2].  If we see the same step value 2x, could that also indicate a process that died/restarted from checkpoint as well?

On the other hand, perhaps users might also intentionally "overwrite" the same last step as a legitimate use case?  I'm not sure, so the PR as-is seems reasonable, just interested in your thoughts.